spec/parser/options_spec.rb

```diff
@@ -368,6 +380,18 @@ def remaining
       it "must not allow the same hash key to be specified multiple times" do
         expect {parse("--attributes", "name:string", "name:integer")}.to raise_error(Thor::MalformattedArgumentError, "You can't specify 'name' more than once in option '--attributes'; got name:string and name:integer")
       end
+
+      it "accepts value when value 't match validator" do
+        create :attributes => Thor::Option.new("attributes", :type => :hash, :validator => proc { |k,v| /name/ === k && /string/ === v}, :validator_desc => 'description')
+        expect { parse("--attributes", "name:string") }.not_to raise_error
```

Comment:
What do you expect to be tested:

* [ ] Option valditaor also works if arguments are given?
* [ ] Adds validator to arguments as well?